### ISSUE TITLE
HTTP/3 doesn't work with .NET 7 + Arch Linux + msquic

### ISSUE THREAD

**Issue by alexandrehtrb** [dotnet/runtime#79520]
### Description

Hello,

I am trying to use HTTP/3 with .NET 7 on Linux, with msquic. I already accomplished to use HTTP/3 with .NET 6 on Linux before, with msquic too.
For some reason, it is not working.
I am using Arch Linux x64 and installing msquic using [this](https://github.com/microsoft/msquic/discussions/2318) script, the only change being that instead of `git checkout main`, I checkout version v2.1.5.
The msquic package gets installed successfully using the script.

### Reproduction Steps

Make a HTTP/3 request using HttpClient. My test URL is https://quic.aiortc.org/.

### Expected behavior

Should work

### Actual behavior

The following exception appears:

```
System.Net.Http.HttpRequestException: Requesting HTTP version 3.0 with version policy RequestVersionExact while unable to establish HTTP/3 connection.
   at System.Net.Http.HttpConnectionPool.ThrowGetVersionException(HttpRequestMessage request, Int32 desiredVersion, Exception inner)
   at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.DecompressionHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
   at Pororoca.Infrastructure.Features.Requester.PororocaRequester.RequestAsync(IPororocaVariableResolver variableResolver, PororocaHttpRequest req, Boolean disableSslVerification, CancellationToken cancellationToken) in /home/alexandre/Projetos/Pororoca/src/Pororoca.Infrastructure/Features/Requester/PororocaRequester.cs:line 37
```

### Regression?

It works with .NET 6 + msquic 1.9.0

### Configuration

`dotnet --info`

```
.NET SDK:
 Version:   7.0.100
 Commit:    e12b7af219

Ambiente de runtime:
 OS Name:     arch
 OS Version:  
 OS Platform: Linux
 RID:         arch-x64
 Base Path:   /home/alexandre/.dotnet/sdk/7.0.100/

Host:
  Version:      7.0.0
  Architecture: x64
  Commit:       d099f075e4

.NET SDKs installed:
  6.0.403 [/home/alexandre/.dotnet/sdk]
  7.0.100 [/home/alexandre/.dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.11 [/home/alexandre/.dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0 [/home/alexandre/.dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.11 [/home/alexandre/.dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0 [/home/alexandre/.dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  DOTNET_ROOT       [/home/alexandre/.dotnet/]

global.json file:
  Not found
```

**Comment by rzikm** [comment]
I have tried with locally built 2.1.5 and I observe no issues.

Can you check `QuicConnection.IsSupported` property? it should be true.

Alternatively, try `ldd <path/to/libmsquic>` to see if the library can be even loaded properly.

**Comment by danmoseley** [comment]
> Requesting HTTP version 3.0 with version policy RequestVersionExact while unable to establish HTTP/3 connection.

I find it hard to parse this message. Should it say

> Unable to establish HTTP/3 connection. Requested HTTP version 3.0 with version policy RequestVersionExact.

**Comment by wfurt** [comment]
I think the problem lives in the script. The `libmsquic.so` should be `libmsquic.so.xx` to match 
https://github.com/microsoft/msquic/blob/d7737be4c112dabb7ae674b09145ada0dfa5fe5e/scripts/make-packages.sh#L151-L155

This changed during MsQuic release as there were breaking changes. For comparison,
```
 dpkg -L libmsquic
/.
/usr
/usr/lib
/usr/lib/x86_64-linux-gnu
/usr/lib/x86_64-linux-gnu/libmsquic.lttng.so.2.1.4
/usr/lib/x86_64-linux-gnu/libmsquic.so.2.1.4
/usr/share
/usr/share/doc
/usr/share/doc/libmsquic
/usr/share/doc/libmsquic/changelog.gz
/usr/lib/x86_64-linux-gnu/libmsquic.so.2

$ ls -al /usr/lib/x86_64-linux-gnu/libmsquic*
-rwxr-xr-x 1 root root 9809904 Oct 14 05:44 /usr/lib/x86_64-linux-gnu/libmsquic.lttng.so.2.1.4
lrwxrwxrwx 1 root root      18 Oct 14 05:59 /usr/lib/x86_64-linux-gnu/libmsquic.so.2 -> libmsquic.so.2.1.4
-rwxr-xr-x 1 root root 6654896 Oct 14 05:45 /usr/lib/x86_64-linux-gnu/libmsquic.so.2.1.4
```

You can simply package it as `libmsquic.so.2' (this is what .NET tried to load) or you can add symlink as Ubuntu does.

**Comment by nibanks** [comment]
@alexandrehtrb feel free to create a PR in MsQuic to check in the script there. Also, if you know any way to automate testing of Arch Linux, we'd take a PR for that as well (in MsQuic)!
